includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Class AMP_Form_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Form_Sanitizer
+ *
+ * Strips and corrects attributes in forms.
+ */
+class AMP_Form_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Tag.
+	 *
+	 * @var string HTML <form> tag to identify and process.
+	 *
+	 * @since 0.2
+	 */
+	public static $tag = 'form';
+
+	/**
+	 * Sanitize the <img> elements from the HTML contained in this instance's DOMDocument.
+	 *
+	 * @since 0.2
+	 */
+	public function sanitize() {
+
+		/**
+		 * Node list.
+		 *
+		 * @var DOMNodeList $node
+		 */
+		$nodes     = $this->dom->getElementsByTagName( self::$tag );
+		$num_nodes = $nodes->length;
+
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $i = $num_nodes - 1; $i >= 0; $i -- ) {
+			$node = $nodes->item( $i );
+			if ( ! $node instanceof DOMElement ) {
+				continue;
+			}
+
+			$method = 'get';
+			if ( $node->hasAttribute( 'method' ) ) {
+				$method = strtolower( $node->getAttribute( 'method' ) );
+			}
+
+			// Get the action URL.
+			if ( ! $node->hasAttribute( 'action' ) ) {
+				$action_url = esc_url_raw( '//' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'] ); // WPCS: ignore. input var okay, sanitization ok.
```

Comment:
@westonruter I don't normally like using `$_SERVER` if I can avoid it, but this is from your suggestion here: https://github.com/Automattic/amp-wp/pull/871#discussion_r162148603
I feel this should be a little cleaner, but can't think how. any suggestions?

Reply:
I think this is right what you have here, except that there should a beautiful `wp_unslash()` call added to it, since `$_SERVER` vars are auto-slashed for back-compat reasons:

```
esc_url_raw( '//' . wp_unslash( $_SERVER['HTTP_HOST'] ) . wp_unslash( $_SERVER['REQUEST_URI'] ) );
```

The one for `HTTP_HOST` is somewhat overkill.

Reply:
agreed on the overkill. but will unslash the request uri.